plugins/inputs/ipvs/README.md

```diff
@@ -5,24 +5,92 @@ metrics about ipvs virtual and real servers.
 
 **Supported Platforms:** Linux
 
-### Configuration:
+## Configuration
 
 ```toml
 [[inputs.ipvs]]
   # no configuration
 ```
 
-### Permissions:
+## Permissions
 
 Assuming you installed the telegraf package via one of the published packages,
 the process will be running as the `telegraf` user. However, in order for this
 plugin to communicate over netlink sockets it needs the telegraf process to be
 running as `root` (or some user with `CAP_NET_ADMIN` and `CAP_NET_RAW`). Be sure
 to ensure these permissions before running telegraf with this plugin included.
 
-### Example Output:
+## Metrics
 
+### Virtual Servers
+
+For virtual servers, this plugin reports the following:
+
+- `ipvs_virtual_server`
+  - tags:
+    - `sched` - the scheduler in use
+    - `netmask` - the mask used for determining affinity
+    - `address_family` - inet/inet6
+    - ONE of `address` + `port` + `protocol` *OR* `fwmark`
+  - fields:
+    - Connections
+    - PacketsIn
+    - PacketsOut
+    - BytesIn
+    - BytesOut
+    - CPS
+    - PPSIn
+    - PPSOut
+    - BPSIn
+    - BPSOut
+
+Each virtual server will contain tags identifying how it was configured, using
+one of `address` + `port` + `protocol` *OR* `fwmark`. This is how one would
+normally configure a virtual server using `ipvsadm`.
+
+### Real Servers
+
+Metrics reported for each `ipvs_real_server`:
+
+- `ipvs_real_server`
+  - tags:
+    - `address`
+    - `port`
+    - `address_family`
+    - ONE of `virtual_address` + `virtual_port` + `virtual_protocol` OR `virtual_fwmark`
+  - fields:
+    - ActiveConnections
+    - InactiveConnections
+    - Connections
+    - PacketsIn
+    - PacketsOut
+    - BytesIn
+    - BytesOut
+    - CPS
+    - PPSIn
+    - PPSOut
+    - BPSIn
+    - BPSOut
+
+Each real server can be identified as belonging to a virtual server using one of
+either `virtual_address + virtual_port + virtual_protocol` OR `virtual_fwmark`
+
+## Example Output
+
+### Virtual servers
+
+Example:
 ```
 ipvs_virtual_server,address=172.18.64.234,address_family=inet,netmask=32,port=9000,protocol=tcp,sched=mh_418 bytes_out=0i,pps_in=0i,pps_out=0i,cps=0i,pkts_in=0i,pkts_out=0i,connections=0i,bytes_in=0i 1540407540000000000
 ipvs_virtual_server,address_family=inet,fwmark=47,netmask=32,sched=mh_418 connections=0i,pkts_in=0i,bytes_out=0i,pps_in=0i,pps_out=0i,pkts_out=0i,bytes_in=0i,cps=0i 1540407540000000000
 ```
+
+### Real servers
+
+Example:
+```
+ipvs_real_server,active_conns=0,address=172.18.64.204,address_family=inet,inactive_conns=0,port=9000,virtual_address=172.18.64.234,virtual_port=9000,virtual_protocol=tcp bytes_out=0i,pps_in=0i,pps_out=0i,cps=0i,connections=0i,pkts_in=0i,pkts_out=0i,bytes_in=0i 1540577170000000000
```

Comment:
Can you regen this so the active_conns are moved to fields?

Reply:
Done